dotnet-desktop-guide/net/winforms/controls/how-to-add-an-event-handler.md

```diff
@@ -0,0 +1,116 @@
+---
+
+# How to handle an event (Windows Forms .NET)
+
+Events for controls (and for forms) are generally set through the Visual Studio Visual Designer for Windows Forms. Setting an event through the Visual Designer is known as handling an event at design-time. You can also handle events dynamically in code, known as handling events at run-time.
+
+In addition to handling events using the Windows Forms Designer in Visual Studio, you can also create an event handler at run-time. An event created at run-time allows you to connect event handlers dynamically based on what your app is currently doing.
+
+[!INCLUDE [desktop guide under construction](../../includes/desktop-guide-preview-note.md)]
+
+## Designer
+
+In Visual Studio, use the Visual Designer to manage handlers for control events. The Visual Designer will generate the handler code and add it to the event for you.
+
+### Set the handler
+
+Use the **Properties** toolbox to add or set the handler of an event:
+
+01. Open the Visual Designer of the form containing the control to change.
+01. Select the control.
+01. Change the **Properties** toolbox mode to **Events** by pressing the events button (:::image type="icon" source="../media/shared/visual-studio-events-button.png" border="false":::).
+01. Find the event you want to add a handler to, for example, the **Click** event:
+
+    :::image type="content" source="media/how-to-add-an-event-handler/visual-studio-properties-events-click.png" alt-text="Visual Studio properties toolbox shown with the events mode enabled and the click event.":::
+
+01. Do one of the following:
+
+    - Double-click the event to generate a new handler, it's blank if no handler is assigned. If it's not blank, this action opens the code for the form and navigates to the existing handler.
+
+    - Use the selection box (:::image type="icon" source="../media/shared/visual-studio-chevron-button.png" border="false":::) to choose an existing handler.
+  
+      The selection box will list all methods that have a compatible method signature for the event handler.
+
+### Clear the handler
+
+To remove an event handler, you can't just delete handler code that is in the form's code-behind file, it's still referenced by the event. Use the **Properties** toolbox to remove the handler of an event:
+
+01. Open the Visual Designer of the form containing the control to change.
+01. Select the control.
+01. Change the **Properties** toolbox mode to **Events** by pressing the events button (:::image type="icon" source="../media/shared/visual-studio-events-button.png" border="false":::).
+01. Find the event containing the handler you want to remove, for example, the **Click** event:
+
+    :::image type="content" source="media/how-to-add-an-event-handler/visual-studio-properties-events-click.png" alt-text="Visual Studio properties toolbox shown with the events mode enabled and the click event.":::
+
+01. Right-click on the event and choose **Reset**.
+
+## Programmatic
```

Comment:
Similar comment to "Designer" H2. Make it more descriptive. i.e. "Handle events programmatically", "Handle events at run-time", "Handle events (code-first)"

Reply:
@luisquintanilla How does this sound? I used the same format on the other article too. I'm trying not to be too wordy on the headings because they're wrapping.